### PR TITLE
Added PerspectiveTransform

### PR DESCRIPTION
**Pull request by radarhere**
Furthers the work of #7684 to help clarify ImageTransform for #7683, by adding code and documentation for the missing ImageTransform - PerspectiveTransform.

This is based on [AffineTransform](https://github.com/python-pillow/Pillow/blob/a786a0551b75ab3e85da1bfad54226faa2336022/src/PIL/ImageTransform.py#L45-L61), since it is similar.
https://github.com/python-pillow/Pillow/blob/a786a0551b75ab3e85da1bfad54226faa2336022/src/libImaging/Geometry.c#L375-L417

If the 7th (g) and 8th (h) values are 0, then `perspective_transform` is identical to `affine_transform`, so`PERSPECTIVE` should be able to do everything that `AFFINE` can.

### CODE REVIEW

File: Tests/test_image_transform.py [hunk 1]
```diff
@@ -16,6 +16,8 @@ def test_sanity(self):
 
         transform = ImageTransform.AffineTransform(seq[:6])
         im.transform((100, 100), transform)
+        transform = ImageTransform.PerspectiveTransform(seq[:8])
+        im.transform((100, 100), transform)
```
Comment (hugovk) [inline Tests/test_image_transform.py:20]: Can we add some correctness tests rather than just a "didn't crash" test?

Comment (radarhere) [reply Tests/test_image_transform.py]: Ok, done

**Review by homm — APPROVED** [review]

**Review by hugovk — APPROVED** [review]
Should we add to release notes?

**Comment by radarhere** [comment]
Ok, sure, I've added release notes.

**Pull request merged**